src/lib/ProjectM36/Client/Simple.hs

```diff
@@ -0,0 +1,79 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+module ProjectM36.Client.Simple (
+  withDBConnection,
+  withTransaction,
+  executeDatabaseContextExpr,
+  executeRelationalExpr,
+  Atom(..),
+  AtomType(..),
+  DBError(..),
+  Attribute(..),
+  C.ConnectionInfo(..),
+  C.PersistenceStrategy(..),
+  C.NotificationCallback(..),
+  C.emptyNotificationCallback,
+  C.DatabaseContextExpr(..),
+  C.RelationalExprBase(..)  
+  ) where
+-- | A simplified, monad-based client for Project:M36 database access.
+import ProjectM36.Error
+import ProjectM36.Base
+import qualified ProjectM36.Client as C
+import Control.Monad.Reader
+import Control.Exception.Base
+
+-- | A union of connection and other errors that can be returned from 'withDBConnection'.
+data DBError = ConnError C.ConnectionError |
+               RelError RelationalError
+               deriving (Eq, Show)
+
+newtype DB a = DB {runDB :: ReaderT (C.SessionId, C.Connection) IO a}
+  deriving (Functor, Applicative, Monad, MonadIO)
```

Comment:
Here you're allowing `liftIO`. Was that intentional? I recall that you weren't certain you wanted it.

Reply:
Sounds good.